ftw/linkchecker/command/checking_links.py

```diff
@@ -0,0 +1,51 @@
+from AccessControl.SecurityManagement import newSecurityManager
+from Testing.makerequest import makerequest
+from zc.relation.interfaces import ICatalog
+from zope.component import getUtility
+from zope.component.hooks import setSite
+from zope.globalrequest import setRequest
+import AccessControl
+
+
+def get_plone_sites_information(app):
+    # make fake request to be able to traverse
+    app = makerequest(app)
+    setRequest(app.REQUEST)
+    overview_view = app.unrestrictedTraverse('plone-overview')
+    plone_sites = overview_view.sites()
+    # create information dictionary for each plone site
+    plone_sites_information = [{
+        'id': plone_site.getId(),
+        'path': '/'.join(plone_site.getPhysicalPath()),
+        'title': plone_site.Title()
+    } for plone_site in plone_sites]
+
+    return plone_sites_information
+
+
+def get_relation_catalog_for_plone_site(app, PloneSiteId):
+    plone_site = app.get(PloneSiteId)
+
+    user = AccessControl.SecurityManagement.SpecialUsers.system
```

Comment:
All the security manager related stuff should be moved out to the `main()` function, that's something you always want to run, and as early as possible. We usually create something like a `setup_plone()` helper function that does that, and returns the appropriately set up Plone site.

Reply:
👍